python/pyarrow/tests/parquet/test_parquet_encryption.py

```diff
@@ -0,0 +1,388 @@
+
+
+@pytest.mark.parquet
+def test_encrypted_parquet_write_kms_error(tempdir):
+    """Write an encrypted parquet, but raise KeyError in KmsClient."""
+    path = tempdir / 'encrypted_table_kms_error.in_mem.parquet'
+    table = DATA_TABLE
+
+    encryption_config = BASIC_ENCRYPTION_CONFIG
+
+    # Empty master_keys_map
+    kms_connection_config = pq.KmsConnectionConfig()
+
+    def kms_factory(kms_connection_configuration):
+        # Empty master keys map will cause KeyError to be raised
+        # on wrap/unwrap calls
+        return InMemoryKmsClient(kms_connection_configuration)
+
+    with pytest.raises(RuntimeError, match="footer_key.*KeyError"):
+        crypto_factory = pq.CryptoFactory(kms_factory)
+        # Write with encryption properties
+        write_encrypted_parquet(path, table, encryption_config,
+                                kms_connection_config, crypto_factory)
+
+
+@pytest.mark.parquet
+def test_encrypted_parquet_write_kms_specific_error(tempdir):
+    """Write an encrypted parquet, but raise KeyError in KmsClient."""
+    path = tempdir / 'encrypted_table_kms_error.in_mem.parquet'
+    table = DATA_TABLE
+
+    encryption_config = BASIC_ENCRYPTION_CONFIG
+
+    # Empty master_keys_map
+    kms_connection_config = pq.KmsConnectionConfig()
+
+    class ThrowingKmsClient(pq.KmsClient):
+        """A KmsClient implementation that throws exception in
+        wrap/unwrap calls
+        """
+
+        def __init__(self, config):
+            """Create an InMemoryKmsClient instance."""
+            pq.KmsClient.__init__(self)
+            self.config = config
+
+        def wrap_key(self, key_bytes, master_key_identifier):
+            raise ValueError("Cannot Wrap Key")
+
+        def unwrap_key(self, wrapped_key, master_key_identifier):
+            raise ValueError("Cannot Unwrap Key")
+
+    def kms_factory(kms_connection_configuration):
+        # Exception thrown in wrap/unwrap calls
+        return ThrowingKmsClient(kms_connection_configuration)
+
+    with pytest.raises(RuntimeError, match="Cannot Wrap Key.*ValueError"):
+        crypto_factory = pq.CryptoFactory(kms_factory)
+        # Write with encryption properties
+        write_encrypted_parquet(path, table, encryption_config,
+                                kms_connection_config, crypto_factory)
+
+
+@pytest.mark.parquet
+def test_encrypted_parquet_write_kms_factory_error(tempdir):
+    """Write an encrypted parquet, but raise ValueError in kms_factory."""
+    path = tempdir / 'encrypted_table_kms_factory_error.in_mem.parquet'
+    table = DATA_TABLE
+
+    encryption_config = BASIC_ENCRYPTION_CONFIG
+
+    # Empty master_keys_map
+    kms_connection_config = pq.KmsConnectionConfig()
+
+    def kms_factory(kms_connection_configuration):
+        raise ValueError('Cannot create KmsClient')
+
+    with pytest.raises(RuntimeError,
+                       match="Cannot create KmsClient.*ValueError"):
```

Comment:
I'm not sure why `RuntimeError` is raised, instead of the original `ValueError`?